apps/angular/crud/src/app/error-handling/global-error-handler.ts

```diff
@@ -0,0 +1,14 @@
+import { ErrorHandler, Injectable, inject } from '@angular/core';
+import { LoadingService } from '../data-access/loading.service';
+
+@Injectable({ providedIn: 'root' })
+export class GlobalErrorHandler implements ErrorHandler {
+  loadingService = inject(LoadingService);
+
+  handleError(error: any): void {
+    console.error('The following error was found:', error);
+    window.alert('There was an error: ' + error);
+    window.location.reload();
+    this.loadingService.stopLoading();
```

Comment:
this is a very dangerous way of handling errors. If a request never success, your application will reload infinitly. 

Why not creating an interceptor to log error, but don't put any logic here. In an entreprise application, errors might be handle differently.

Reply:
@tomalaforge Instead of reloading the page, what about if I route to an error page showing the error. This page can have possible actions the user can take. I added this class for errors that are not http related.

Reply:
Yes this is better. You can create a 404 page inside your routing config and route to this page when there are some errors